JetStreamCompose/jetstream/src/main/java/com/google/jetstream/data/repositories/MovieRepositoryImpl.kt

```diff
@@ -33,132 +33,167 @@ import com.google.jetstream.data.util.StringConstants.Movie.Reviewer.DefaultCoun
 import com.google.jetstream.data.util.StringConstants.Movie.Reviewer.DefaultRating
 import com.google.jetstream.data.util.StringConstants.Movie.Reviewer.FreshTomatoes
 import com.google.jetstream.data.util.StringConstants.Movie.Reviewer.ReviewerName
+import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.flow
-import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+import kotlinx.coroutines.withContext
 import kotlinx.serialization.json.Json
 import javax.inject.Inject
 
-class MovieRepositoryImpl @Inject constructor(assetsReader: AssetsReader) : MovieRepository {
+class MovieRepositoryImpl @Inject constructor(
+    private val assetsReader: AssetsReader
+) : MovieRepository {
 
-    private val mostPopularMovies: List<MoviesResponseItem> =
-        assetsReader.getJsonDataFromAsset(StringConstants.Assets.MostPopularMovies).map {
-            Json.decodeFromString<List<MoviesResponseItem>>(it)
-        }.getOrDefault(emptyList())
+    private val mostPopularMovies = flow {
```

Comment:
I assume you're having `flow`s here, because you're later combining it in viewmodels and combining `flow` and `suspend` is not easy.

Because you're not subscribed to any data changes, I suggest you can transform the UI in a different manner, using `suspend` so you can easily combine it:

`HomeScreenViewModel`::
```kotlin
    private val _uiState = MutableStateFlow<HomeScreenUiState>(HomeScreenUiState.Loading)
    val uiState: StateFlow<HomeScreenUiState> = _uiState

    init {
        viewModelScope.launch {
            val featuredMovieList = async { movieRepository.getFeaturedMovies() }
            val trendingMovieList = async { movieRepository.getTrendingMovies() }
            val top10MovieList = async { movieRepository.getTop10Movies() }
            val nowPlayingMovieList = async { movieRepository.getNowPlayingMovies() }

            _uiState.value = HomeScreenUiState.Ready(
                featuredMovieList.await(),
                trendingMovieList.await(),
                top10MovieList.await(),
                nowPlayingMovieList.await(),
            )
        }
    }
```

Reply:
Thanks for the suggestion. Actually, your comment totally makes sense, and I would keep using flows for the following two reasons

- I would keep the room for the sample code to handle updates happening on MovieRepository. 
- Flow allow me to write more readable code.